cmd/cluster/sre_operators/listCmd.go

```diff
@@ -0,0 +1,475 @@
+  clients to fetch the latest versions of each operator from its respective repository.
+	`
+	repositoryBranch = "production"
+)
+
+func newCmdList(streams genericclioptions.IOStreams, client client.Client) *cobra.Command {
+	opts := &sreOperatorsListOptions{
+		kubeCli:   client,
+		IOStreams: streams,
+	}
+
+	listCmd := &cobra.Command{
+		Use:               "list",
+		Short:             "List the current and latest version of SRE operators",
+		Long:              sreOperatorsListDescription,
+		Example:           sreOperatorsListExample,
+		Args:              cobra.NoArgs,
+		DisableAutoGenTag: true,
+		Run: func(cmd *cobra.Command, args []string) {
+			util.CheckErr(opts.checks(cmd))
+			output, _ := opts.ListOperators(cmd)
+			util.CheckErr(opts.printText(output))
+		},
+	}
+
+	listCmd.Flags().BoolVar(&opts.short, "short", false, "Exclude fetching the latest version from repositories for faster output")
+	listCmd.Flags().BoolVar(&opts.outdated, "outdated", false, "Filter to only show operators running outdated versions")
+	listCmd.Flags().BoolVar(&opts.noHeaders, "no-headers", false, "Exclude headers from the output")
+	listCmd.Flags().StringVar(&opts.operator, "operator", "", "Filter to only show the specified operator.")
+
+	return listCmd
+}
+
+// Command validity check
+func (ctx *sreOperatorsListOptions) checks(cmd *cobra.Command) error {
+	if _, err := config.GetConfig(); err != nil {
+		return util.UsageErrorf(cmd, "could not find KUBECONFIG, please make sure you are logged into a cluster")
+	}
+	if ctx.outdated && ctx.short {
+		return util.UsageErrorf(cmd, "cannot use both --short and --outdated flags together")
+	}
+	return nil
+}
+
+// Print output in table format
+func (ctx *sreOperatorsListOptions) printText(opList []sreOperator) error {
+	p := printer.NewTablePrinter(ctx.IOStreams.Out, 20, 1, 3, ' ')
+
+	if opList == nil {
+		return nil
+	}
+
+	if !ctx.noHeaders {
+		p.AddRow([]string{Bold})
+		header := []string{"NAME", "CURRENT", "EXPECTED", "STATUS", "CHANNEL" + RestoreColor}
+		if ctx.short {
+			header = []string{"NAME", "CURRENT", "STATUS", "CHANNEL" + RestoreColor}
+		}
+		p.AddRow(header)
+	}
+
+	sort.Slice(opList, func(i, j int) bool {
+		return opList[i].Name < opList[j].Name
+	})
+
+	for _, op := range opList {
+		if ctx.outdated && op.Current == op.Expected {
+			continue
+		}
+		row := []string{op.Name, op.Current, op.Status, op.Channel}
+		if !ctx.short {
+			row = []string{op.Name, op.Current, op.Expected, op.Status, op.Channel}
+			if op.Current != op.Expected {
+				row = []string{Red + Bold + op.Name, Gap + op.Current, Gap + op.Expected, Gap + op.Status, Gap + op.Channel + RestoreColor}
+			}
+		}
+		p.AddRow(row)
+	}
+	if err := p.Flush(); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+type operatorResult struct {
+	Operator sreOperator
+	Error    error
+}
+
+func (ctx *sreOperatorsListOptions) ListOperators(cmd *cobra.Command) ([]sreOperator, error) {
+
+	listOfOperators := []string{
```

Comment:
This seems to be repeated in both `describe` and `list` files. Can we move the definition to either of the files or a separate file entirely

Reply:
Refactored to be delcared only once in listCmd file, thanks for the suggestion